clippy_lints/src/manual_clamp.rs

```diff
@@ -103,6 +105,24 @@ struct ClampSuggestion<'tcx> {
     hir_with_ignore_attr: Option<HirId>,
 }
 
+impl<'tcx> ClampSuggestion<'tcx> {
+    /// This function will return true if and only if you can demonstrate at compile time that min
+    /// is less than max.
+    fn min_less_than_max(&self, cx: &LateContext<'tcx>) -> bool {
+        let max_type = cx.typeck_results().expr_ty(self.params.max);
+        let min_type = cx.typeck_results().expr_ty(self.params.min);
+        if max_type != min_type {
+            return false;
+        }
+        let max = constant(cx, cx.typeck_results(), self.params.max);
+        let min = constant(cx, cx.typeck_results(), self.params.min);
+        let cmp = max
+            .zip(min)
+            .and_then(|(max, min)| Constant::partial_cmp(cx.tcx, max_type, &min, &max));
+        cmp.is_some_and(|cmp| cmp != Ordering::Greater)
```

Comment:
What do you think about:

```suggestion
        if let Some(max) = constant(cx, cx.typeck_results(), self.params.max)
            && Some(min) = constant(cx, cx.typeck_results(), self.params.min)
        {
            Constant::partial_cmp(cx.tcx, max_type, &min, &max) != Ordering::Greater
        } else {
            false
        }
```

?

Reply:
I like it. I had to make some minor tweaks to make the compiler happy, but I've implemented your suggestion in aecdb921a.